inc/vcf/validator.hpp

```diff
@@ -185,12 +189,16 @@ namespace ebi
                            ValidationLevel validationLevel,
                            std::vector<std::unique_ptr<ebi::vcf::ReportWriter>> &outputs);
 
-    bool is_compressed_file(const std::string &source,
-                            const std::vector<char> &line);
+    std::string get_compressed_file(const std::string &source,
```

Comment:
alright, sorry to ask for another renaming, but I don't think the current names are good enough. what about `get_compression`, `get_compression_from_extension` and `get_compression_from_magic_num`?